### PR TITLE
README.md: add instructions for softdevice

### LINKED ISSUE #128: mention softdevice in readme
Another issue that held me up this weekend was this library requires softdevice already be present, but doesnt mention it anywhere. Maybe just copy paste https://github.com/sandeepmistry/arduino-nRF5#flashing-a-softdevice to here?

### PR DESCRIPTION
**Pull request by jacobrosenthal**
Closes  #128

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,23 @@ cd ~/Documents/mpide/libraries/
 git clone https://github.com/sandeepmistry/arduino-BLEPeripheral BLEPeripheral
 ```
 
+### Flashing a SoftDevice
+
```
Comment (sandeepmistry) [inline README.md]: I'm not too keen to duplicate stuff from the nRF5 read me here, what do you think of having a link to it?

Also, these steps are only needed when you also use the nRF5 core. Other scenarios include cores like the original RBL ones or when using the an nRF8001 shield with an Uno, Zero, etc.

Maybe we need an nRF5x core section?

File: README.md [hunk 2]
```diff
@@ -111,6 +128,9 @@ See [API.md](API.md).
 ## Examples
 See [examples](examples) folder.
 
+## Troubleshooting
+* If you see ```fatal error: ble_gatts.h: No such file or directory``` it actually means you havent selected your softdevice, and the BLEPeripheral SDK files are not available to the Verify
+
```
Comment (sandeepmistry) [inline README.md]: Again, also only applies for nRF5x core

**Comment by jacobrosenthal** [comment]
ble_gatts.h Is odd in the core too though, because it has nothing to
do with core..

But lets just say we that under softdevice in core, then just add a
link there talking about softdevice for nordic devices?

**Comment by sandeepmistry** [comment]
> ble_gatts.h Is odd in the core too though, because it has nothing to
do with core..

Right, but that's the only place we can allow people to select the right version and custom include paths.


> But lets just say we that under softdevice in core, then just add a
link there talking about softdevice for nordic devices?

Sure, that works.

**Pull request merged**